scene/main/node.h

```diff
@@ -187,7 +187,9 @@ class Node : public Object {
 
 #ifdef TOOLS_ENABLED
 	friend class SceneTreeEditor;
+	friend class GLTFDocument;
```

Comment:
I'm curious as to whether _validate_node_name should just be made public instead.

Reply:
Is anything else using it? Do we want other things to use it?

Reply:
I don't think anything else is using it today, but I'm also not sure if there are other cases where sanitization is being performed (maybe in the other import modules?). In my mind it'd be better for consistency's sake if we'd have everything that generates a Node and needs fine control over naming would leverage `_validate_node_name` so any future modifications to expand/reduce the restricted characters would be applied uniformly. 

Similarly since AnimationPlayer adds a couple additional restrictions, I think we would ideally have a similar public static `_validate_animation_name` that can perform animation-specific cleanup.

Reply:
I would try hard to find at least one other use or invent a use because if nothing uses it, we don't need to expose.

Reply:
> I would try hard to find at least one other use or invent a use because if nothing uses it, we don't need to expose.

You mean making `_validate_node_name` public specifically? On the one hand it feels odd to make `GLTFDocument` a full friend class just to get access to one method (I assume `SceneTreeEditor` does more interesting Node manipulation), but if there's a general trend towards being conservative about making utility methods public that's fine as well.

Reply:
Yes please make `_validate_node_name` public (and remove the `_` private prefix), and expose it in `_bind_methods`.

Reply:
@akien-mga  Big 👍 on making it public (done, and removed the `STE` `friend` declaration as it seems it was really only used to access name validation). 
Wrt. exposing in `_bind_methods`; it feels like this particular method would provide an atypical GDScript API as it potentially modifies the given String parameter and returns a `bool` indicating if the parameter was modified. Would it be better to expose a wrapping method that returns an `Array` (maybe `[was_valid, validated_name]`) instead?

Reply:
Ah right, that type of API doesn't work so well for scripting langs. I think this could be refactored into `String Node::validate_node_name(const String p_string)` (i.e. returns a modified string instead of changing in place), though maybe the current form was made this way to avoid allocations in performance critical code?

Reply:
@reduz suggests adding and exposing `String String::validate_node_name(const String &p_string)` instead, and using that in Node and other relevant code that needs to sanitize a node name.

Reply:
Glancing at the blame I suspect it was just to facilitate the use in `STE` which wants to display an error message if the node name had to be sanitized: 6758b6c1317bbdabd04a2c9f8dc16876429b6bfb
If that's the case it seems like it'd be fine to have it return the string and then do a (admittedly expensive) equality check on the original and returned string in that one spot. 

I'm happy to make that change if you think it's worthwhile?

Reply:
Sorry, missed the later comment regarding moving the validation to `String`. 
My initial reaction is that approach feels like it smears `Node` concerns into a potentially unexpected place (naively if I was looking for node validation and saw a `Node` class I'd assume it was there). That said, `String` does have things like `simplify_path` that arguably would be appropriate in `_File`.

Reply:
Poking around a bit more I see that there doesn't appear to be a way to expose a static method via ClassDB so if there's a strong desire to make this bindable, moving it to `String` with the signature: `String String::validate_node_name() const` (where it returns a copy of the string after sanitizing invalid characters) seems like it'd be the most straightforward approach, though it still feels slightly out of place to me.

@akien-mga I'll wait for direction from you/@reduz before going that path, but in the meantime I've pushed a new commit with a script friendly API in case there's a way to bind `static` methods that I've missed. If it's not necessary to bind this method then it'd probably make sense to revert this latest commit to avoid the performance penalty of copying the string.

Reply:
Yeah the scripting bindings don't support static methods, so we have to work it around for now. That's why we tend to (ab)use the String API for some File and Node checks. If we do implement support for static methods, then it would indeed make sense to have this under Node, but for the time being the String approach is the simplest one for a binding.

Reply:
Makes sense, thanks for the guidance! I've made the change (and added documentation and a basic unit test).